src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Rules
+{
+    public class XamlRuleTests
+    {
+        [Theory]
+        [MemberData(nameof(GetFileItemRules))]
+        public void FileRulesShouldMatchNone(string ruleName, string fullPath)
+        {
+            // Special case for Folder rule which hasn't been split yet, but is in the Items folder. But also its completely different.
+            if (ruleName.Equals("Folder", StringComparison.Ordinal))
+            {
+                return;
+            }
+            // No need to check None against None
+            if (ruleName.Equals("None", StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            string noneFile = Path.Combine(fullPath, "..", "None.xaml");
+
+            XmlDocument none = LoadXamlRule(noneFile);
+            XmlDocument rule = LoadXamlRule(fullPath);
+
+            // First fix up the Name as we know they'll differ.
+            rule.DocumentElement.Attributes["Name"].Value = "None";
+
+            AssertXmlEqual(none, rule);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetBrowseObjectItemRules))]
+        public void BrowseObjectRulesShouldMatchNone(string ruleName, string fullPath)
+        {
+            // Special case for Folder rule which hasn't been split yet, but is in the Items folder. But also its completely different.
+            if (ruleName.Equals("Folder", StringComparison.Ordinal))
+            {
+                return;
+            }
+            // No need to check None against None
+            if (ruleName.Equals("None", StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            string noneFile = Path.Combine(fullPath, "..", "None.BrowseObject.xaml");
+
+            XmlDocument none = LoadXamlRule(noneFile);
+            XmlDocument rule = LoadXamlRule(fullPath);
+
+            // First fix up the Name and DisplayName as we know they'll differ.
+            rule.DocumentElement.Attributes["Name"].Value = "None";
+            rule.DocumentElement.Attributes["DisplayName"].Value = "General";
+
+            AssertXmlEqual(none, rule);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllRules))]
+        public void NonVisiblePropertiesShouldntBeLocalized(string ruleName, string fullPath)
```

Comment:
Do we need another test that verifies visible properties have `DisplayName` (and possibly other fields)?

Reply:
Yes, and many more to boot

Reply:
Would be good to capture your ideas for such tests in a tracking issue, if they're not already.

Reply:
I'm generally tracking everything in #4105